### PR TITLE
Update linting rules to Vue3

### PR DESCRIPTION
**Pull request by guergana**
This PR adds `vue/vue3-strongly-recommended` to check for vue linting violations in vue3

Bug: T351644

### CODE REVIEW

**Review by chukarave — COMMENTED** [review]
So nice to see so many lint fixes at once :smile:

File: app/Http/Controllers/ResultsController.php [hunk 1]
```diff
@@ -65,10 +65,10 @@ public function index(MismatchGetRequest $request, WikibaseAPIClient $wikidata):
         $props = array_merge(
             [
                 'user' => $user,
-                'item_ids' => $requestedItemIds,
+                'itemIds' => $requestedItemIds,
```
Comment (chukarave) [inline app/Http/Controllers/ResultsController.php:68]: Did eslint complain about a php file or was it a manual fix for consistency's sake?

Comment (chukarave) [reply app/Http/Controllers/ResultsController.php:68]: but did it lint this PHP file? I thought eslint only lints JS

Comment (guergana) [reply app/Http/Controllers/ResultsController.php:68]: these are passed as Props to the Results page here: https://github.com/wmde/wikidata-mismatch-finder/pull/836/files#diff-f0e07e4099d24711f323a8ef5a958c7723c8bdaca46379665835500790de1c3fR245

If it doesnt have the same name some tests fail.

Comment (chukarave) [reply app/Http/Controllers/ResultsController.php:68]: Ah, I see. that sounds fine

**Review by chukarave — APPROVED** [review]
Looks good to me :+1:

**Pull request merged**